tests/test_docker.py

```diff
@@ -180,6 +180,15 @@ def test_build_with_buildargs(self):
         client = docker.from_env(
             version='auto',
             timeout=180)
-        self.assertEqual(
-            'somevalue',
-            client.containers.run(im, remove=True).decode())
```

Comment:
According to docs this should just returned output, it seems they may have introduced a bug.

Reply:
It's the `remove=True` part that causes an issue, if you tell docker to remove a container once finished, it will also remove the ability to query logs, so you end up with a problem that if the API behind the run doesn't attach quickly enough (some delay in system execution) the container may be removed by the time it requests the output.

Looks to me like it's a simple race that goes away if you skip setting `remove=True`, so I suspect they don't test with it set to `True`.

Reply:
It should be behaviour similar to docker run, where logs are not fetched after run but collected during runtime.

Reply:
Collecting the logs during runtime still requires that you capture them quickly enough, only if they were capturing the output directly from when `container.start()` was called would they guarantee capturing the output. See https://github.com/docker/docker-py/blob/master/docker/models/containers.py#L801-L809 the problem is that the output is not being captured so there is a risk that when the log output is requested, docker daemon may have already deleted the container.